src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -739,6 +737,29 @@ public void emitNodeMetrics(long currWindowStartTime, MetricsDB metricsDB) throw
     }
   }
 
+  /**
+   * An example value is this: current_time:1566413987194 StartTime:1566413987194 ItemCount:359
+   * IndexName:nyc_taxis ShardID:25 Primary:true Each pair is separated by new line and the key and
+   * value within each pair is separated by ":" This function just parses the string and generates
+   * the map as such.
+   *
+   * @param eventValue The value input to the helper function.
+   * @return Returns a map of key value pairs
+   */
+  static Map<String, String> extractEntryData(String eventValue) {
+    String[] lines = eventValue.split(System.lineSeparator());
```

Comment:
Why are we not using `MetricsParesr.extractEntryData` ?

Reply:
I have removed the file MetricsParser.java altogether. None of the functions in it were being used and only this method was being used in a few places.

Reply:
Awesome !